### ISSUE TITLE
postgres defined types are not recognized

### ISSUE THREAD

**Issue by jagypus** [dbcodeio/public#164]
I have defined a number of types such as:
CREATE TYPE application AS (
    id UUID,
    created_at TIMESTAMPTZ,
    updated_at TIMESTAMPTZ,
}

which are then referenced by functions such as:
CREATE OR REPLACE FUNCTION api.applications_list()
 RETURNS SETOF application

these types do not show up in the database viewer - neither under 'Types' nor as the return type of the function (in tree view), in fact, the tree view for this function  

![Image](https://github.com/user-attachments/assets/653e69e6-ea32-4cf5-b566-8c1e55df6d0f)

**Comment by jagypus** [comment]
if I run this SQL then I see the types I have defined:

SELECT 
    n.nspname as schema_name,
    t.typname as type_name,
    string_agg(
        a.attname || ' ' || pg_catalog.format_type(a.atttypid, a.atttypmod),
        ', ' ORDER BY a.attnum
    ) as structure
FROM pg_type t
JOIN pg_namespace n ON n.oid = t.typnamespace
JOIN pg_class c ON c.oid = t.typrelid
JOIN pg_attribute a ON a.attrelid = c.oid
WHERE n.nspname = 'api'
AND t.typtype = 'c'  -- Only composite types
AND a.attnum > 0     -- Skip system attributes
GROUP BY n.nspname, t.typname
ORDER BY schema_name, type_name;

**Comment by mikeburgh** [comment]
Thanks for raising all of these, should be easy to pick up trigger and void… will need to do some testing on the composite types, especially around editing columns using those types (not sure if that’s doable in pg, or if they are only usable in procs, I will do some testing and see)

**Comment by jagypus** [comment]
I don't think you can use these types (or the composite types) in tables, I think they are only for function return types (and maybe input parameters, but i've not tried that) - i'm just using them for return types.

**Comment by mikeburgh** [comment]
Turns out you can use them in tables 😉 

```sql
CREATE TYPE inventory_item AS (
    name            text,
    supplier_id     integer,
    price           numeric
);

CREATE TABLE on_hand (
    item      inventory_item,
    count     integer
);

INSERT INTO on_hand VALUES (ROW('fuzzy dice', 42, 1.99), 1000);
```

So picked them up, and added support for them, will be fixed in todays release

**Comment by mikeburgh** [comment]
Fixed in 1.5.9, let me know if they are still causing you issues, but should be sorted now.
